pkg/ibmvpc/sg_analysis.go

```diff
@@ -323,14 +323,35 @@ func (sga *SGAnalyzer) AllowedConnectivity(target string, isIngress bool) *commo
 }
 
 // rulesInConnectivity list of SG rules contributing to the connectivity
-func (sga *SGAnalyzer) rulesInConnectivity(target string, isIngress bool) []int {
+func (sga *SGAnalyzer) rulesInConnectivity(target string, conn *common.ConnectionSet, isIngress bool) ([]int, error) {
 	analyzedConns, ipb := sga.getAnalyzedConnsIPB(target, isIngress)
 	for definedTarget, rules := range analyzedConns.contribRules {
 		if ipb.ContainedIn(definedTarget) {
-			return rules
+			if conn != nil { // connection not part of the query
+				contained, err := conn.ContainedIn(analyzedConns.allowedconns[definedTarget])
```

Comment:
containment of conns here and intersection of conns in `getRulesRelevantConn` -- add some documentation about the difference (here it is the top level result per target, and only if the input connection is fully contained in the allowed conn, we consider this as "allowed" and look for contributing rules, even if they only intersect and not fully contained).

Reply:
done